service/src/main/java/com/blog/controller/AuthorRestController.java

```diff
@@ -0,0 +1,81 @@
+package com.blog.controller;
+
+import com.blog.Author;
+import com.blog.Post;
+import com.blog.service.AuthorService;
+import com.blog.service.PostService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import javax.validation.ValidationException;
+import java.util.List;
+
+@RestController
+@RequestMapping("/authors")
+public class AuthorRestController {
+
+    private AuthorService authorService;
+    private PostService postService;
+
+    @Autowired
+    public AuthorRestController(AuthorService authorService, PostService postService) {
+        this.authorService = authorService;
+        this.postService = postService;
+    }
+
+    @GetMapping("")
+    @ResponseStatus(value = HttpStatus.OK)
+    public List<Author> getAllAuthors() {
+        return authorService.getAllAuthors();
+    }
+
+    @GetMapping("/{id}")
+    @ResponseStatus(value = HttpStatus.OK)
+    public Author getAuthorById(@PathVariable(value = "id") Long userId) {
+        return authorService.getAuthorById(userId);
+    }
+
+    @GetMapping("/{id}/posts")
+    @ResponseStatus(value = HttpStatus.OK)
+    public List<Post> getAllPostsByAuthorId(@PathVariable(value = "id") Long userId) {
+        return postService.getAllPostsByAuthorId(userId);
+    }
+
+    @GetMapping("/login/{login}")
+    @ResponseStatus(value = HttpStatus.OK)
+    public Author getAuthorByLogin(@PathVariable(value = "login") String login) {
+        return authorService.getAuthorByLogin(login);
+    }
+
+    @PostMapping("")
+    @ResponseStatus(value = HttpStatus.CREATED)
+    public Long addAuthor(@Valid @RequestBody Author author, BindingResult result) {
+        if (result.hasErrors()) {
+            throw new ValidationException(result.getFieldErrors().toString());// TODO: протестировать!!!
+        } else {
+            return authorService.addAuthor(author);
+        }
+    }
+
+    @PutMapping("")
```

Comment:
According to REST approaches it might be better to use /author/{id} URL for updating existing entity.

Reply:
There're 2 cases possible
1. Check that URL id equals to object id
2. Just set URL ID in object and use it as is.